internal/pool/pool.go

```diff
@@ -93,7 +93,7 @@ func (p *Pool) dequeue() *task {
 			p.mu.Unlock()
 
 			select {
-			case <-time.After(t.deadline.Sub(time.Now())):
+			case <-time.After(time.Until(t.deadline)):
```

Comment:
I remember the `time.After` idiom being an issue in earlier Golang versions. As long as we're Go 1.23+, this won't be a resource leak. ([`time.After` docs](https://cs.opensource.google/go/go/+/go1.25.1:src/time/sleep.go;l=202))